### PR TITLE
Update docker deployment

### LINKED ISSUE #781: Improve documentation on how to deploy a Taipy application with Docker
**Description**
Some information are not given for the deployment with Docker.

[Page](https://docs.taipy.io/en/latest/manuals/run-deploy/deploy/docker/production/)
image

The link doesn't help in knowing what is to change. It also lacks what has to be done to the Python script.

On this [page](https://docs.taipy.io/en/latest/manuals/run-deploy/deploy/docker/development/), it is not told that host should be `host='0.0.0.0'` in the run of the application for example.

**Expected change**
The goal would be to add more information.

### PR DESCRIPTION
**Pull request by florian-vuillemot**
closes #781

### CODE REVIEW

File: docs/manuals/run-deploy/deploy/docker/production.md [hunk 1]
```diff
@@ -7,14 +7,9 @@
...
-The following Dockerfile contains the minimum configuration settings for you to deploy your application to
-production.
+The following Dockerfile contains the minimum configuration settings to deploy your application to production.
 
-This template assumes that you provide a `requirements.txt` file with all the Python dependencies of your
-application.
-
-Make sure you update the Docker command `CMD` with the appropriate file name and Flask settings for your
-application as describe in [Running a Taipy application](../../run/index.md).
+This template assumes that you provide a `requirements.txt` file with all the Python dependencies of your application and that your application entry point is the file `main.py.`
```
Comment (jrobinAV) [inline docs/manuals/run-deploy/deploy/docker/production.md]:
```suggestion
This template assumes that you provide a `requirements.txt` file with all the Python
dependencies of your application and that your application entry point is the file `main.py.`
```

File: docs/manuals/run-deploy/deploy/docker/production.md [hunk 2]
```diff
@@ -31,17 +26,18 @@ USER taipy
...
+RUN python -m pip install -r requirements.txt
 
 # Install your application
 COPY . .
-RUN python -m pip install -r requirements.txt
 
 # Start up command
-ENTRYPOINT [ "gunicorn", "-k", "geventwebsocket.gunicorn.workers.GeventWebSocketWorker", "-w", "1", "--bind=0.0.0.0:5000", "--timeout", "1800" ]
-CMD [ "<main>:<app>" ]
+ENTRYPOINT [ "python", "main.py", "-P", "5000", "-H", "0.0.0.0", "--no-reloader" ]
```
Comment (jrobinAV) [inline docs/manuals/run-deploy/deploy/docker/production.md:41]: why not `taipy run` ?

File: docs/manuals/run-deploy/deploy/docker/production.md [hunk 3]
```diff
@@ -57,11 +53,7 @@ CMD [ "<main>:<app>" ]
 
 ## Production ready Docker Compose
 
-It is not recommended to expose Gunicorn to the Internet. It is better to let [Nginx](https://nginx.org)
-handle [this responsibility](https://docs.gunicorn.org/en/stable/deploy.html).
-
-Beside your `docker-compose.yaml`, create a file named `nginx.conf` with
-[the following content](./nginx.conf).
+Exposing your application directly to the Internet is not recommended. Let [Nginx](https://nginx.org) handle this responsibility better for security and reliability. Consequently, beside your `docker-compose.yaml`, create a file named `nginx.conf` with [the following content](./nginx.conf).
```
Comment (jrobinAV) [inline docs/manuals/run-deploy/deploy/docker/production.md]:
```suggestion
Exposing your application directly to the Internet is not recommended. Letting
[Nginx](https://nginx.org) handle the responsibility is better for security and reliability.
Consequently, besides your `docker-compose.yaml`, create a file named `nginx.conf` with 
[the following content](./nginx.conf).
```

File: docs/manuals/run-deploy/deploy/docker/production.md (hunk 2 above)
Comment (florian-vuillemot) [reply docs/manuals/run-deploy/deploy/docker/production.md:41]: Good point! Do we all agree that this is the right thing to do?

Comment (jrobinAV) [reply docs/manuals/run-deploy/deploy/docker/production.md:41]: I would say yes. I can't see any reason not to.

Comment (FabienLelaquais) [reply docs/manuals/run-deploy/deploy/docker/production.md:41]: One reason could be 'keeping simple things simple'?

**Review by FabienLelaquais — APPROVED** [review]
👍

**Pull request merged**